src/test/service.test.ts

```diff
@@ -1143,4 +1143,82 @@ test(
   })
 );
 
+test(
+  'service in watch mode persists when non-cascading dependency restarts or fails',
+  // service1
+  //    |
+  //    v
+  // service2 (restarts and fails)
+  timeout(async ({rig}) => {
+    const service1 = await rig.newCommand();
+    const service2 = await rig.newCommand();
```

Comment:
Unbelievably minor nit and can be discarded. Naming these `parentService` and `childService` could help make it easier to follow the test. I kept losing track of whether `1` or `2` was the child.
Also please ignore this comment.

Reply:
I can +1 that I did have to look at the config being written to tell which one was the dependent. But also, super minor since it is indeed right there.

Reply:
Done